proposals/0033-timely-vote-credits.md

```diff
@@ -0,0 +1,389 @@
+   larger form unless the vote account has sufficient lamports to cover
+   the new higher rent exempt minimum.  The new size of vote accounts will
+   be 3762 bytes, a 31 byte increase on the old size of 3731 bytes.  The
+   rent exempt minimum for vote accounts will thus increase from 0.02685864 SOL
+   to 0.0270744 SOL.  This is an additional 0.00021576 SOL per vote account.
+   Given roughly 2,000 vote accounts currently on mainline, the maximum SOL
+   needed to increase all vote accounts to the required new minimum is
+   0.43152 SOL.  The true value will certainly be less than this because
+   some vote accounts will already be above the new minimum.  In any case,
+   the author of the SIMD will send any additional SOL needed to increase
+   rent exempt minimum of all vote accounts, in the weeks leading up to
+   the enablement of this feature.  By the time the feature is enabled, the
+   vast majority of vote accounts will be funded to the new rent exempt
+   minimum.
+
+   For any vote account not sufficiently funded, the old VoteState version
+   will be written.  Since the old VoteState does not record latencies,
+   vote account deserialization will assign the default value of 0 to
+   the Lockout latency field, which will result in the minimum vote credits
+   awarded for each vote for that vote account.  This provides economic
+   incentive to validators to sufficiently fund their vote accounts to allow
+   the updated VoteStateVersion to be serialized into them.
+
+   To summarize the issues surrounding the Vote account updates:
+
+   - A vote account will only be updated to the new version after the
+     feature is enabled, and only if the vote account includes enough
+     SOL to satisfy the new rent exempt minimum.
+   - If a vote account is not updated, then it will continue to work, but
+     because its vote latencies will not be recorded, and will default to
+     0, the vote account will earn minimum credits for votes until
+     updated
+
+   There is no danger of breaking vote processing; the only danger is to
+   validators who do not supply sufficient rent exempt minimum, and that
+   danger is limited to reduced earnings through lower vote credits.  There
+   is no risk to cluster consensus.
+
+2. A second change will be made, with a separate feature, that will enable
+   the recording of the latency values for votes in the latency field of
+   Lockout, and the subsequent use of this latency value to compute vote
+   credits awarded.
+
+   Once this feature is enabled, vote accounts will earn credits based on
+   their vote latencies.  The maximum possible credits earnings per epoch
+   will increase from 432,000 to 3,456,000.  Each validator will continue to
+   earn vote rewards based on their fraction of total vote credits earned
+   across the cluster, and validators which earn more credits through
+   reduced vote latencies will earn a larger percentage of the total
+   rewards.
+
+## A Note About the Tower
```

Comment:
I think this section can be folded into the previous, talking about the backwards compatibility of the tower.

Reply:
It can be, but why?  It's meant to provide documentation about a specific and complex and confusing aspect to the implementation.

Reply:
SIMDs are client agnostic, any references to the Solana Labs implementation are better suited for the PR in that repo

Reply:
SIMDs are client agnostic, any references to the Solana Labs implementation are better suited for the PR in that repo